tutorials/generative/realism_diversity_metrics/realism_diversity_metrics.py

```diff
@@ -0,0 +1,356 @@
+
+# ## Compute FID
+
+# The FID measures the distance between the feature vectors from the real images and those obtained from generated images. In order to compute the FID the images need to be passed into a pre-trained network to get the desired feature vectors. Although the FID is commonly computed using the Inception network, here, we used a pre-trained version of the RadImageNet to calculate the feature space. Lower FID scores indicate that the images are more similar, with a perfect score being 0 indicating that the two groups of images are identical.
+
+radnet = torch.hub.load("Warvito/radimagenet-models", model="radimagenet_resnet50", verbose=True)
+radnet.to(device)
+radnet.eval()
+
+# Here, we will load the real and generate synthetic images from noise and compute the FID of these two groups of images. Because we are generating the synthetic images on this code snippet the entire cell will take about 6 mins run and most of the this time is spent in generating the images. The loading bars show how long it will take to complete the image generation for each mini-batch.
+
+# +
+fid_scores = []
+unet.eval()
+
+for step, x in enumerate(val_loader):
+    # Get the real images
+    real_images = x["image"].to(device)
+
+    # Generate some synthetic images using the defined model
+    n_synthetic_images = len(x['image'])
+    noise = torch.randn((n_synthetic_images, 1, 64, 64))
+    noise = noise.to(device)
+    scheduler.set_timesteps(num_inference_steps=25)
+
+    with torch.no_grad():
+        syn_images, intermediates = inferer.sample(input_noise=noise, diffusion_model=unet,
+                                                  scheduler=scheduler,save_intermediates=True,
+                                                  intermediate_steps=100)
+
+        # Get the features for the real data
+        real_eval_feats = get_features(real_images)
+
+        # Get the features for the synthetic data
+        synth_eval_feats = get_features(syn_images)
+
+        fid = FIDMetric()
+        fid_res = fid(synth_eval_feats.to(device), real_eval_feats.to(device))
+        fid_scores.append(fid_res)
+
+# -
+
+fid_scores = torch.stack(fid_scores)
+print(f"FID Score: {fid_scores.mean().item():.4f} +- {fid_scores.std().item():.4f}")
+
+# Plot 3 examples from the synthetic data
+fig, ax = plt.subplots(nrows=1, ncols=3)
+for image_n in range(3):
+    ax[image_n].imshow(syn_images[image_n, 0, :, :].cpu(), cmap="gray")
+    ax[image_n].axis("off")
+
+# # Compute MMD
+
+# MMD (Maximum Mean Discrepancy) is a distance metric used to measure the similarity between two probability distributions. This metric maps the samples from each distribution to a high-dimensional feature space and calculates the distance between the mean of the features of each distribution. A smaller MMD value indicates a better match between the real and generated distributions. It is often used in combination with other evaluation metrics to assess the performance of a generative model.
```

Comment:
Here, I would comment that the realism of the LDM depends on the quality/realism of the AE reconstructions. For this reason you are computing the MMD between the original images and the image_recon